src/mito2/src/engine/alter_test.rs

```diff
@@ -573,6 +574,80 @@ async fn test_alter_column_fulltext_options() {
     check_region_version(&engine, region_id, 1, 3, 1, 3);
 }
 
+#[tokio::test]
+async fn test_alter_region_ttl_options() {
+    common_telemetry::init_default_ut_logging();
+
+    let mut env = TestEnv::new();
+    let listener = Arc::new(AlterFlushListener::default());
+    let engine = env
+        .create_engine_with(MitoConfig::default(), None, Some(listener.clone()))
+        .await;
+
+    let region_id = RegionId::new(1, 1);
+    let request = CreateRequestBuilder::new().build();
+
+    env.get_schema_metadata_manager()
+        .register_region_table_info(
+            region_id.table_id(),
+            "test_table",
+            "test_catalog",
+            "test_schema",
+            None,
+        )
+        .await;
+    let region_dir = request.region_dir.clone();
+    engine
+        .handle_request(region_id, RegionRequest::Create(request))
+        .await
+        .unwrap();
+    let engine_cloned = engine.clone();
+    let alter_ttl_request = RegionAlterRequest {
+        schema_version: 0,
+        kind: AlterKind::SetRegionOptions {
+            options: vec![SetRegionOption::TTL(Duration::from_secs(500))],
+        },
+    };
+    let alter_job = tokio::spawn(async move {
+        engine_cloned
+            .handle_request(region_id, RegionRequest::Alter(alter_ttl_request))
+            .await
+            .unwrap();
+    });
+
+    alter_job.await.unwrap();
+
+    let check_ttl = |engine: &MitoEngine, expected: &Duration| {
+        let current_ttl = engine
+            .get_region(region_id)
+            .unwrap()
+            .version()
+            .options
+            .ttl
+            .unwrap();
+        assert_eq!(*expected, current_ttl);
+    };
+    // Verify the ttl.
+    check_ttl(&engine, &Duration::from_secs(500));
+
+    // Reopen region.
+    let engine = env.reopen_engine(engine, MitoConfig::default()).await;
+    engine
+        .handle_request(
+            region_id,
+            RegionRequest::Open(RegionOpenRequest {
+                engine: String::new(),
+                region_dir,
+                options: HashMap::default(),
+                skip_wal_replay: false,
+            }),
+        )
+        .await
+        .unwrap();
+    // Verify the ttl again after engine restart.
+    // check_ttl(&engine, &Duration::from_secs(500));
```

Comment:
Hi @WenyXu, the ttl verify will fail after region restart. The restarted region had `None` as ttl.

Reply:
https://github.com/lyang24/greptimedb/blob/5faf2f483a2399960cd10df7380d517591977f66/src/mito2/src/region/options.rs#L102-L149

It's expected that you need to open the region with the specified options, including the `ttl`.

i.e.,
```rust
   let options = HashMap::from([("ttl".to_string(), "500s".to_string())]);
    engine
        .handle_request(
            region_id,
            RegionRequest::Open(RegionOpenRequest {
                engine: String::new(),
                region_dir,
                options,
                skip_wal_replay: false,
            }),
        )
        .await
        .unwrap();
```

Reply:
I see thanks, i will just remove the reopen engine part of the test as hardcode an option do not provide much value for this test.